Sources/MoSoCore/MoSoUser.swift

```diff
@@ -2,12 +2,19 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+import Foundation
+
 // TODO: this very simple implementation will change
 public class MoSoUser {
     public let username: String
     public let identifier: String
     @KeychainStorage public var session: MoSoSession?
-    @KeychainStorage public var pocketSession: MoSoSession?
+    @KeychainStorage public var pocketSession: MoSoSession? {
+        didSet {
+            let hasToken = pocketSession != nil
+            NotificationCenter.default.post(name: .userAuthDidChange, object: nil, userInfo: ["hasToken": hasToken])
```

Comment:
An extreme nit-pick here as a suggestion would be to update the name to something like `pocketSessionDidChange`, and the user info could contain the session, or an `isAuthenticated` bool.

Reply:
I chose to not do that as only by actually trying to perform a request will we know if the token is valid and the user is actually authed.
The existence of a token does not mean we have auth. Only that we have a token :D

Reply:
And I kept the notification name generic for when we move away from Pocket auth later.